tests/hil/tests/ota/test.c

```diff
@@ -168,6 +172,24 @@ static void test_block_ops(void)
     }
 }
 
+static void test_restart(void)
+{
+
+    if (golioth_sys_sem_take(restart_test_begin_sem, 0))
+    {
+        golioth_client_stop(client);
+        golioth_sys_msleep(3000);
+        GLTH_LOGI(TAG, "Restarting Client");
+        golioth_sys_sem_give(restart_pending_sem);
+        golioth_client_start(client);
+    }
+    else if (golioth_sys_sem_take(restart_pending_sem, 0))
+    {
+        GLTH_LOGI(TAG, "Restart successful");
+        golioth_sys_sem_give(restart_test_begin_sem);
```

Comment:
```suggestion
```
We don't need to reset any semaphores for this test. We want each to run once and only once, to prevent restarting the test when the artifact is rolled back in the teardown phase of the test.